speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DataCompressionMenu.cpp

```diff
@@ -0,0 +1,239 @@
+#define COMPRESSION_SCREEN_NAME "DataCompressionMenu"
+
+#define MAX_COMPRESSION_LEVEL 49
+
+// GUI screen handles
+static void* s_scrHandle = nullptr;
+static void* s_prevHandle = nullptr;
+
+// GUI settings Id's
+int m_compressionButtonList;
+
+// InterventionType
+DataCompressionType m_dataCompressionType;
+
+// Custom compression level
+int m_customCompressionLevel;
+int m_customCompressionLevelControl;
+
+// Apply Button
+int m_applyButtonComp;
+
+/// @brief         Loads the settings from the config file
+/// @param p_param A handle to the parameter file
+static void LoadSettingsFromFile(void* p_param)
+{
+    m_dataCompressionType = std::stoi(GfParmGetStr(p_param, PRM_DATA_COMPRESSION, GFMNU_ATTR_SELECTED, "1"));
+    m_customCompressionLevel = std::stoi(GfParmGetStr(p_param, PRM_CUSTOM_COMPRESSION, GFMNU_ATTR_TEXT, nullptr));
+}
+
+/// @brief Makes sure all visuals display the internal values
+static void SynchronizeControls()
+{
+    GfuiRadioButtonListSetSelected(s_scrHandle, m_compressionButtonList, static_cast<int>(m_dataCompressionType));
+
+    char buf[32];
+    sprintf(buf, "%d", m_customCompressionLevel);
+    GfuiEditboxSetString(s_scrHandle, m_customCompressionLevelControl, buf);
+}
+
+/// @brief Loads default settings
+static void LoadDefaultSettings()
+{
+    m_dataCompressionType = GfuiRadioButtonListGetSelected(s_scrHandle, m_compressionButtonList);
+    m_customCompressionLevel = std::stoi(GfuiEditboxGetString(s_scrHandle, m_customCompressionLevelControl));
+}
+
+/// @brief Loads (if possible) the settings; otherwise, the control's default settings will be used
+static void LoadSettings()
+{
+    char buf[512];
+    sprintf(buf, "%s%s", GfLocalDir(), COMPRESSION_FILEPATH);
+    if (GfFileExists(buf))
+    {
+        void* param = GfParmReadFile(buf, GFPARM_RMODE_STD);
+        // Initialize settings with the retrieved xml file
+        LoadSettingsFromFile(param);
+        SynchronizeControls();
+        return;
+    }
+    LoadDefaultSettings();
+}
+
+/// @brief Saves the settings to a file
+static void SaveSettingsToFile()
+{
+    std::string dstStr(COMPRESSION_FILEPATH);
+    char dst[512];
+    sprintf(dst, "%s%s", GfLocalDir(), dstStr.c_str());
+    void* readParam = GfParmReadFile(dst, GFPARM_RMODE_REREAD | GFPARM_RMODE_CREAT);
+
+    // Write data compression type
+    char val[32];
+    sprintf(val, "%d", m_dataCompressionType);
+    GfParmSetStr(readParam, PRM_DATA_COMPRESSION, GFMNU_ATTR_SELECTED, val);
+
+    // Save custom compression to xml file
+    char buf[32];
+    sprintf(buf, "%d", m_customCompressionLevel);
+    GfParmSetStr(readParam, PRM_CUSTOM_COMPRESSION, GFMNU_ATTR_TEXT, buf);
+
+    // Write queued changes
+    GfParmWriteFile(nullptr, readParam, COMPRESSION_SCREEN_NAME);
+}
+
+/// @brief Saves the settings so the mediator (or future instances) can access them
+static void SaveSettings()
+{
+    int compressionLevel;
+    switch (m_customCompressionLevel)
```

Comment:
I'm pretty sure this value should be the type and not the value, otherwise the type would not get used anywhere.
```suggestion
    switch (m_dataCompressionType)
```